libraries/chain/include/eosio/chain/finality_core.hpp

```diff
@@ -33,9 +33,9 @@ struct qc_claim_t
 
 struct core_metadata
 {
-   block_num_type  last_final_block_num;
-   block_num_type  final_on_strong_qc_block_num;
-   block_num_type  latest_qc_claim_block_num;
+   block_num_type  last_final_block_num {0};
+   block_num_type  final_on_strong_qc_block_num {0};
+   block_num_type  latest_qc_claim_block_num {0};
```

Comment:
I believe this is unneeded, int members are default initialized to 0. 
![image](https://github.com/AntelopeIO/leap/assets/1496907/4f9e55f0-d54b-439e-a382-90b1e0d9f52c)

Reply:
I was bitten by not initializing an integer member not long ago during local development. Plus explicit initialization marks intension clearer?

Reply:
I like the safety of explicit initialization. Shouldn't make any difference as our current only creation is:
```cpp
return core_metadata {
      .last_final_block_num = new_last_final_block_num,
      .final_on_strong_qc_block_num = new_final_on_strong_qc_block_num,
      .latest_qc_claim_block_num = most_recent_ancestor_with_qc.block_num,
   };
```

Reply:
OK, but I don't see how it is safer to re-specify the implicit initialization.

Reply:
Not sure why you are claiming implicit initialization for ints.
https://godbolt.org/z/4Gd986Ea9

Reply:
I wonder if that's a gcc bug? weirdly, if you uncomment line 35 [here](https://godbolt.org/z/3vEdMc955) these are zero-initialized. Other compilers as well. 
In this particular case I am still not convinced the explicit initializers are needed, but I guess it is fair to argue the opposite!

Reply:
Same for clang: https://godbolt.org/z/EWsGf15K6
Uncomment and run multiple times. It is indeterminate, so run it a few times.

Reply:
https://stackoverflow.com/questions/706030/are-uninitialized-struct-members-always-set-to-zero